### PR TITLE
feat(react-hook,card): integrate reat hook form in checkbox radio switch and create toggle card

### PR DESCRIPTION
**Pull request by AzizSandid**
## What does this do?

Integrates React Hook Form with checkboxes, radio buttons, switches, and creates a toggle card. 
## Why did you do this?

To improve form handling and validation in the application by leveraging React Hook Form's capabilities for better state management and performance.

### CODE REVIEW

File: src/shared/components/checkbox-input.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { useController } from 'react-hook-form';
+
+import { Checkbox } from './checkbox';
+type Props<T extends FieldValues> = {
+  className?: string;
+  accessibilityLabel: string;
+  label: string;
+  name: Path<T>;
+  control: Control<T>;
+  rules?: any;
```
Comment (MohamedMegdich1) [inline src/shared/components/checkbox-input.tsx:12]: Why are you using any type?

File: src/shared/components/checkbox.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { MotiView } from 'moti';
 import React, { useCallback } from 'react';
-import { Pressable, type PressableProps } from 'react-native';
+import { Pressable, type PressableProps, View } from 'react-native';
```
Comment (MohamedMegdich1) [inline src/shared/components/checkbox.tsx:3]: why not import from shared/component

File: src/shared/components/radio-input.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { useController } from 'react-hook-form';
+
+import { Radio } from './radio';
+type Props<T extends FieldValues> = {
+  className?: string;
+  accessibilityLabel: string;
+  label: string;
+  name: Path<T>;
+  control: Control<T>;
+  rules?: any; // Adjust the type according to your validation rules
```
Comment (MohamedMegdich1) [inline src/shared/components/radio-input.tsx:12]: the same

File: src/shared/components/switch-input.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import { useController } from 'react-hook-form';
+
+import { Switch } from './switch';
+type Props<T extends FieldValues> = {
+  className?: string;
+  accessibilityLabel: string;
+  label: string;
+  name: Path<T>;
+  control: Control<T>;
+  rules?: any; // Adjust the type according to your validation rules
```
Comment (MohamedMegdich1) [inline src/shared/components/switch-input.tsx:12]: the same

File: src/shared/components/toggle-card.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+type CardProps<T extends FieldValues> = {
+  title?: string;
+  image?: string;
+  className?: string;
+  classNameText?: string;
+  svgComponent?: React.ComponentType;
+  name: Path<T>;
+  control: Control<T>;
+  rules?: any; // Adjust the type according to your validation rules
```
Comment (MohamedMegdich1) [inline src/shared/components/toggle-card.tsx:17]: the same

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/shared/components/checkbox-input.tsx (hunk 1 above)
Comment (AzizSandid) [reply src/shared/components/checkbox-input.tsx]: fixed

File: src/shared/components/checkbox.tsx (hunk 2 above)
Comment (AzizSandid) [reply src/shared/components/checkbox.tsx]: fixed

File: src/shared/components/radio-input.tsx (hunk 3 above)
Comment (AzizSandid) [reply src/shared/components/radio-input.tsx]: fixed

File: src/shared/components/toggle-card.tsx (hunk 5 above)
Comment (AzizSandid) [reply src/shared/components/toggle-card.tsx]: fixed

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: src/shared/components/select.tsx [hunk 6]
```diff
@@ -6,13 +6,13 @@ import { FlashList } from '@shopify/flash-list';
...
 import type { FieldValues } from 'react-hook-form';
 import { useController } from 'react-hook-form';
-import { Platform, TouchableOpacity, View } from 'react-native';
-import { Pressable, type PressableProps } from 'react-native';
 import type { SvgProps } from 'react-native-svg';
 import Svg, { Path } from 'react-native-svg';
 import { tv } from 'tailwind-variants';
 
 import { CaretDown } from '@/assets/icons';
+import { Platform, TouchableOpacity, View } from '@/shared/components';
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/shared/components/select.tsx:14]: fixed

File: src/shared/components/switch-input.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import type {
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/shared/components/switch-input.tsx:2]: fixed

File: src/shared/components/toggle-card.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import type {
+  Control,
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/shared/components/toggle-card.tsx:3]: fixed

File: src/shared/components/checkbox-input.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import type {
```
[the remark this answers is not in the document]
Comment (AzizSandid) [reply src/shared/components/checkbox-input.tsx:2]: fixed

**Review by MohamedMegdich1 — APPROVED** [review]

**Pull request merged**